packages/suite/src/components/suite/modals/TransactionDetail/components/AmountDetails/index.tsx

```diff
@@ -175,30 +175,65 @@ const AmountDetails = ({ tx, isTestnet }: Props) => {
                     </>
                 )}
                 {/* AMOUNT */}
-                <AmountRow
-                    firstColumn={<Translation id="AMOUNT" />}
-                    secondColumn={<FormattedCryptoAmount value={amount} symbol={assetSymbol} />}
-                    thirdColumn={
-                        showHistoricalRates && (
-                            <FiatValue
-                                amount={amount}
-                                symbol={assetSymbol}
-                                source={tx.rates}
-                                useCustomSource
-                            />
-                        )
-                    }
-                    fourthColumn={
-                        showFiat && (
-                            <FiatValue
-                                amount={amount}
-                                symbol={assetSymbol}
-                                tokenAddress={tokenTransfer?.address}
-                            />
-                        )
-                    }
-                    color="light"
-                />
+                {tx.targets.length > 0 && (
+                    <AmountRow
+                        firstColumn={<Translation id="AMOUNT" />}
+                        secondColumn={
+                            <FormattedCryptoAmount value={tx.amount} symbol={tx.symbol} />
+                        }
+                        thirdColumn={
+                            showHistoricalRates && (
+                                <FiatValue
+                                    amount={amount}
+                                    symbol={assetSymbol}
+                                    source={tx.rates}
+                                    useCustomSource
+                                />
+                            )
+                        }
+                        fourthColumn={
+                            showFiat && (
+                                <FiatValue
+                                    amount={amount}
+                                    symbol={assetSymbol}
+                                    tokenAddress={tokenTransfer?.address}
+                                />
+                            )
+                        }
+                        color="light"
+                    />
+                )}
+                {tx.tokens.map((t, i) => (
+                    <AmountRow
+                        // eslint-disable-next-line react/no-array-index-key
+                        key={i}
```

Comment:
If I am not wrong, `tx.tokens` is array of all "token transfers" (term borrowed from ethereum) in the transaction. With ethereum you could send only 1 token per tx, in cardano you can send multiple tokens to multiple outputs. So you could even send same token to multiple outputs. So you would need identifier composed of token id and index of the output which is not available here and I think it would be the same as using index from .map anyway.